### PR TITLE
Refac/UI ux

### PR DESCRIPTION
**Pull request by espressoGoddess**
## What I did:
* This fixes the loading gif size and location so that it is visible on all screen sizes
* This adds a link to the logo to take the user back to dashboard
* This adds logo and navbar to `_app.tsx` and removes them from the other pages
* This updates the layout and width of the navbar on different screen sizes
* This fixes the logo to the top of the page and adds a dark background only if scrolling to prevent text being on top
* This makes the `Form` have a form with an `onSubmit` handler instead of onClick for the button
* This adds a slider with arrows on each section of the form so users see that there are more options to see
* This refactors the form component for better readability
* This removes the Form css file and utilizes tailwind instead
* This makes html more semantic
* This adds the word 'moon' if its a new moon or full moon for better UX
* This fixes a bug on the redirect to registration form (for new users)
* This updates types on RenderDashboard component
* This updates UserData type since userID is not needed on the client side anymore
* Overall the responsiveness is better

## Testing Notes: 
* Did not touch testing

## Did this break anything?
- [X]  No
- [ ]  Yes
- [ ]  Potentially (explain)

## Type of change
- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [X]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.

## Checklist:
- [ ]  If this code needs to be tested, all tests are passing
- [X]  The code runs locally
- [X]  The code is DRY. If it's not, I tried my best
- [X]  I reviewed my code before pushing

## What's next?
* Tests are failing, but not related to this branch
* Feel free to let me know if there's something you think could benefit from adjustment related to the UI/UX

### CODE REVIEW

**Comment by espressoGoddess** [comment]
@spacenate can you review this?

File: src/components/Form.tsx [hunk 1]
```diff
@@ -51,7 +78,18 @@ const Form = ({ entryDate, isAuthorized, data, updateEntryDate, selections, setS
     return () => setError(null);
   }, [isAuthorized, entryDate, data, router, setSelections]);
 
-  const postForm = async () => {
+  useEffect(() => {
+    const handleScroll = () => {
+      setScrolling(window.scrollY > 0);
```
Comment (spacenate) [inline src/components/Form.tsx:83]: This "window.scrollY > 0" logic (and conditional BG color) shows up in two places. If only just to help reduce the scope of responsibility of this Form component, wdyt about abstracting this into a separate component? Perhaps a ScrollYContext, and/or OpaqueOnScrollY component (with conditional BG)...

File: src/pages/_app.tsx [hunk 2]
```diff
@@ -6,29 +6,49 @@ import React from 'react';
...
-        logIn={logIn}
-        entryDate={entryDate}
-        updateEntryDate={updateEntryDate}
-        selections={selections}
-        setSelections={setSelections}
-      />
-    </>
-  );
-}
+  if(asPath === '/registrationform' || asPath === '/') {
```
[the remark this answers is not in the document]
Comment (spacenate) [reply src/pages/_app.tsx:26]: There's a minor display/alignment issue on the profile page (at mobile/tablet viewports), it seems the logo didn't exist on that page previously

File: src/components/CelestialLogo.tsx [hunk 3]
```diff
@@ -1,10 +1,29 @@
...
+    };
+    window.addEventListener('scroll', handleScroll);
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, []);
+
   return (
-    <Image className='ml-5' width={300} height={100} alt="Logo" src={logoImg} />
+    <div className={`fixed top-0 left-0 z-10 transition-opacity max-sm:w-full ${scrolling ? 'bg-deepblue' : ''}`}>
```
Comment (spacenate) [inline src/components/CelestialLogo.tsx:22]: What do you think about a short CSS transition on the background? Toggling between two different RGBA colors (one with opacity 0) could be a nice effect 🙂

**Review by spacenate — APPROVED** [review]
All of the changes here look reasonable to me 🙌 Aim for slightly smaller/more focused PRs next time, and keep up the good work!

**Pull request merged**